src/Compilers/Core/Portable/InternalUtilities/OneOrMany.cs

```diff
@@ -228,6 +232,49 @@ public bool Any<TData>(Func<T, TData, bool> predicate, TData data)
         public ImmutableArray<T> ToImmutable()
             => this.HasOne ? ImmutableArray.Create(_one) : _many;
 
+        public T[] ToArray()
+            => this.HasOne ? new[] { _one } : _many.ToArray();
+
+        public bool SequenceEqual(OneOrMany<T> other, IEqualityComparer<T>? comparer = null)
+        {
+            comparer ??= EqualityComparer<T>.Default;
+
+            if (Count != other.Count)
+            {
+                return false;
+            }
+
+            return HasOne ? comparer.Equals(_one, other[0]) :
+                   other.HasOne ? comparer.Equals(_many[0], other._one) :
```

Comment:
ok.  def seems wonky from what i undersatnd the last time i looked at this type.  i *though* that if you only had one element, it was always in `_one` not in `_many` as an array of length 1.

Reply:
ok.  instead of indexing into _one or _many.  i would literally just iterate from 0 to count, and just use the indexing operator on `this` or `other`.

Reply:
@CyrusNajmabadi Might be worth updating the constructor that takes an ImmutableArray to fetch the element from the array if it has a single one and store it in the field rather then holding on the array.

Reply:
i def agree with that.